src/main/java/com/faforever/client/chat/UserInfoWindowController.java

```diff
@@ -368,56 +382,92 @@ private void enterAchievementsLoadedState() {
     achievementsPane.setVisible(true);
   }
 
-  public void ladder1v1ButtonClicked() {
-    loadStatistics(KnownFeaturedMod.LADDER_1V1);
+  public void onRatingTypeChange() {
+    CompletableFuture<Void> statisticsFuture = switch (ratingTypeComboBox.getValue()) {
+      case LADDER_1V1 -> loadStatistics(KnownFeaturedMod.LADDER_1V1);
+      default -> loadStatistics(KnownFeaturedMod.FAF);
+    };
+    statisticsFuture.thenRun(() -> Platform.runLater(this::plotPlayerRatingGraph));
   }
 
   private CompletableFuture<Void> loadStatistics(KnownFeaturedMod featuredMod) {
     return statisticsService.getRatingHistory(featuredMod, player.getId())
-        .thenAccept(ratingHistory -> Platform.runLater(() -> plotPlayerRatingGraph(ratingHistory)))
+        .thenAccept(ratingHistory -> ratingData = ratingHistory)
         .exceptionally(throwable -> {
           // FIXME display to user
           log.warn("Statistics could not be loaded", throwable);
           return null;
         });
   }
 
-  @SuppressWarnings("unchecked")
-  private void plotPlayerRatingGraph(List<RatingHistoryDataPoint> dataPoints) {
-    List<XYChart.Data<Integer, Integer>> values = dataPoints.stream()
-        .map(datapoint -> new Data<>(dataPoints.indexOf(datapoint), RatingUtil.getRating(datapoint)))
+  public void plotPlayerRatingGraph() {
+    OffsetDateTime afterDate = switch (timePeriodComboBox.getValue()) {
+      case LAST_MONTH -> OffsetDateTime.of(LocalDateTime.now().minusMonths(1), ZoneOffset.UTC);
+      case LAST_YEAR -> OffsetDateTime.of(LocalDateTime.now().minusYears(1), ZoneOffset.UTC);
+      case ALL_TIME -> OffsetDateTime.of(LocalDateTime.MIN, ZoneOffset.UTC);
```

Comment:
Same here if you put that in the enums adding a new ratingTimePeriod becomes super easy